container-images/tcib/base/os/tempest/run_tempest.sh

```diff
@@ -18,15 +56,10 @@ if [ ! -z ${USE_EXTERNAL_FILES} ]; then
     cp ${TEMPEST_PATH}clouds.yaml $HOME/.config/openstack/clouds.yaml
 fi
 
-if [ -f ${TEMPEST_PATH}profile.yaml ]; then
-    PROFILE_ARG="--profile ${TEMPEST_PATH}profile.yaml"
-fi
-
 tempest init openshift
 
 pushd $TEMPEST_DIR
-
-discover-tempest-config --os-cloud $OS_CLOUD --debug --create identity.v3_endpoint_type public ${PROFILE_ARG}
```

Comment:
--os-cloud $OS_CLOUD
^^ if we wanna maintain the same logic, if we remove it from here, we need to set this here: we need to set this in https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/roles/tempest/vars/main.yml so that it remains a default value (but this way it'll be customizable if needed)

--debug
^^ we need to set this in https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/roles/tempest/vars/main.yml

--create
^^ we need to set this in https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/roles/tempest/vars/main.yml

identity.v3_endpoint_type public
^^ this is ok to remove as it's set ci-framework by default via profile.yaml (that's the only tempestconf interface that ci-framework uses to interact with temepstconf) -> https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/roles/tempest/vars/main.yml

Reply:
ah, ok, I see, I missed that you set true by default on lines 10 and 14 so `--debug` and `--create` are covered, the logic is not changed. 

I'd personally avoid setting default values in this script but that can be changed later (e.g. after this is merged, we can propose changes to ci-framework where we can set --debug and --create by default and after that is merged we could remove the default values here from lines 10 and 14). The same applies on the `identity.v3_endpoint_type public` override.

I agree with your comments about OS_CLOUD.